### DISCUSSION TITLE
How can I use multiple IsGranted attribute on a symfony controller ?

### DISCUSSION THREAD

**Discussion by talha-core** [symfony/symfony#54696] [Q&A] [answered]
I have a controller on which I have 2 routes created:

```php
    #[Route('/announcement/create', name: 'app_admin_restock_announcement_create'), IsGranted(RestockVoter::CREATE_RESTOCK)]
    #[Route('/announcement/edit/{id}', name: 'app_admin_restock_announcement_edit'), IsGranted(RestockVoter::EDIT_RESTOCK, 'restockQueue')]
    public function detail(
        ?RestockQueue $restockQueue,
        ...
```

The problem is user who can `create` can **not** `edit`.

When I debug on route `app_admin_restock_announcement_create` it goes to my custom voter and returns `true`. But at the same time it also return `false` and doesn't go into the controller.

How can I manage this? If I don't want to create separate controllers.

**This is a temporary solution:**
```php
if (is_null($restockQueue)) {
   $this->denyAccessUnlessGranted(RestockVoter::CREATE_RESTOCK);
} else {
   $this->denyAccessUnlessGranted(RestockVoter::EDIT_RESTOCK, $restockQueue);
}
```

But I want it done only using PHP attributes.

**Comment by stof** [comment]
If you put multiple `IsGranted` attributes on a controller method, they are **all** applied when it gets called.

Note that those `IsGranted` attributes are not tied to a route but to a controller (the fact that you group them with route attributes is purely a code style thing but it has no actual grouping effect on the way PHP parses attributes)

**Reply by talha-core** [reply]
So in other words, there is no another solution with my coding style except call `denyAccessUnlessGranted` inside the controller, right ?

**Reply by stof** [reply, marked as answer]
If you want to keep a single controller, you indeed cannot use attribute-based access control checks if you need 2 different checks depending on the case where it is called.
